### PR TITLE
Sanjo/vault upgrade

### PR DESCRIPTION
**Pull request by SanjoDeundiak**

### CODE REVIEW

File: implementations/rust/vault/src/ffi/types.rs [hunk 1]
```diff
@@ -1,144 +1,49 @@
...
-            SecretKeyType::Buffer(_) => SecretKey::Buffer(a),
-            SecretKeyType::Aes128 => SecretKey::Aes128(*array_ref![a, 0, 16]),
-            SecretKeyType::Aes256 => SecretKey::Aes256(*array_ref![a, 0, 32]),
-            SecretKeyType::P256 => SecretKey::P256(*array_ref![a, 0, 32]),
-            SecretKeyType::Curve25519 => SecretKey::Curve25519(*array_ref![a, 0, 32]),
-        };
-        Ok(s)
-    }
-}
+use std::convert::TryInto;
```
Comment (mikelodder7) [inline implementations/rust/vault/src/ffi/types.rs:3]: Nice removing code is always good.

**Comment by SanjoDeundiak** [comment]
@mrinalwadhwa rust changes to vault API touched elixir

**Comment by SanjoDeundiak** [comment]
@robinbudd a lot of changes here...

**Review by mikelodder7 — APPROVED** [review]

**Review by mrinalwadhwa — APPROVED** [review]
the elixir changes look good 👍

**Pull request merged**